solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -1414,10 +1267,17 @@ private void doProcessGroupedDistributedSearchFirstPhase(ResponseBuilder rb, Que
         .setSearcher(searcher);
 
     for (String field : groupingSpec.getFields()) {
+      final int topNGroups;
+      Query query = cmd.getQuery();
+      if (query instanceof AbstractReRankQuery){
+        topNGroups = cmd.getOffset() + ((AbstractReRankQuery)query).getReRankDocs();
+      } else {
+        topNGroups = cmd.getOffset() + cmd.getLen();
+      }
       topsGroupsActionBuilder.addCommandField(new SearchGroupsFieldCommand.Builder()
           .setField(schema.getField(field))
           .setGroupSort(groupingSpec.getGroupSort())
-          .setTopNGroups(cmd.getOffset() + cmd.getLen())
```

Comment:
Is it reRanking with respect to the "start" param in the solr query?
Looks like the original LTR plugin is supposed to reRank top K irrespective of the offset.
ref.http://lucene.472066.n3.nabble.com/Learning-to-Rank-LTR-with-grouping-td4366691i20.html#a4387929

Reply:
True, this currently reranks the top `start +  rerankDocuments `, I'll fix it. Thank you